pkg/cloud/aws/ssm/ssm-operations.go

```diff
@@ -138,36 +139,74 @@ func CheckInstanceInformation(experimentsDetails *experimentTypes.ExperimentDeta
 			return stacktrace.Propagate(err, "failed to check target instance(s) status")
 		}
 		instanceIDList = experimentsDetails.TargetInstanceIDList
-
 	}
+
 	sesh := common.GetAWSSession(experimentsDetails.Region)
 	ssmClient := ssm.New(sesh)
-	for _, ec2ID := range instanceIDList {
-		res, err := ssmClient.DescribeInstanceInformation(&ssm.DescribeInstanceInformationInput{})
+	foundInstances := make(map[string]bool)
+	input := &ssm.DescribeInstanceInformationInput{}
+
+	var err error
+	maxRetries := 5
+	maxRetryDuration := time.Second * 30
+	startTime := time.Now()
```

Comment:
NIT: Formatting for better readability.

```suggestion
        var (
 	    foundInstances make(map[string]bool)
	    input &ssm.DescribeInstanceInformationInput{}
	    err error
	    maxRetries = 5
	    maxRetryDuration = time.Second * 30
	    startTime = time.Now()
      )
```